pkg/sources/git/git.go

```diff
@@ -189,8 +189,12 @@ func (s *Source) Chunks(ctx context.Context, chunksChan chan *sources.Chunk) err
 		return errors.New("invalid connection type for git source")
 	}
 
-	for i, u := range s.conn.Directories {
-		s.SetProgressComplete(i, len(s.conn.Repositories), fmt.Sprintf("Repo: %s", u), "")
+	for _, u := range s.conn.Directories {
+		s.Counter.IncTotal()
+		go func() {
+			defer s.Counter.IncSuccess()
+		}()
```

Comment:
I think the `defer s.Counter.IncSuccess()` should probably go at the end of the for loop?

Reply:
You mean not use the defer and just do s.Counter.IncSuccess()?

Reply:
Yeah, or wrap the entire loop body in a function to defer.

Since we have a defer in a go func here I think it'll just immediately execute currently.